src/theory/strings/array_solver.cpp

```diff
@@ -125,14 +127,33 @@ void ArraySolver::checkTerms(const std::set<Node>& termSet)
         continue;
       }
       // for update terms, also check the inverse inference
-      checkTerm(t, true);
+      checkInv = true;
     }
     else if (k != SEQ_NTH)
     {
       continue;
     }
+
+    // Ignore congruent terms
```

Comment:
Can we just use `BaseSolver::isCongruent` https://github.com/cvc5/cvc5/blob/master/src/theory/strings/base_solver.cpp#L707?  I think this should have the same effect.